tools/probes/vulkan_coopmat_probe.c

```
@@ -0,0 +1,137 @@
+        .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2, .pNext = &s16};
+    vkGetPhysicalDeviceFeatures2(pds[i], &f2);
+    printf("    features: cooperativeMatrix=%d shaderFloat16=%d "
+           "storageBuffer16BitAccess=%d\n",
+           cmf.cooperativeMatrix, f16f.shaderFloat16, s16.storageBuffer16BitAccess);
+
+    if (has_cm && getProps) {
+      uint32_t cn = 0;
+      getProps(pds[i], &cn, NULL);
+      VkCooperativeMatrixPropertiesKHR *cps = calloc(cn, sizeof *cps);
+      for (uint32_t j = 0; j < cn; j++)
+        cps[j].sType = VK_STRUCTURE_TYPE_COOPERATIVE_MATRIX_PROPERTIES_KHR;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)